R/cf.R

```diff
@@ -623,15 +671,52 @@ add.cf <- function(cf1, cf2, a = 1.0, b = 1.0) {
   stopifnot(all(dim(cf1$cf) == dim(cf2$cf)))
   stopifnot(cf1$Time == cf2$Time)
 
+  if( inherits(cf1, 'cf_boot') | inherits(cf2, 'cf_boot') ){
+    res_compat <- resampling_is_compatible(cf1, cf2)
+    if( !all(unlist(res_compat)) ){
+      print(res_compat)
+      stop("Samples were found to be incompatible in add.cf!\n")
+    }
+  }
+
   cf <- cf1
   cf$cf <- a*cf1$cf + b*cf2$cf
 
   if( has_icf(cf1) | has_icf(cf2) ){
     stopifnot( has_icf(cf1) & has_icf(cf2) )
+    stopifnot( all(dim(cf1$icf) == dim(cf2$icf) ) )
     cf$icf <- a*cf1$icf + b*cf2$icf
   }
 
-  cf <- invalidate.samples.cf(cf)
+  # now reconstruct the bootstrap samples
+  invalidate.samples.cf(cf)
```

Comment:
This statement does not have effect, right?

Reply:
The idea, as elsewhere in the code, was to ensure a completely clean slate before re-adding the updated bootstrap samples via the constructor.

Reply:
I realize that part. But without the `cf <-` assignment that line is without an effect due to the pass-b-promise nature of R.

Reply:
Now I see what you mean, my brain didn't even parse that the assignment had been removed...